metal/errors.go

```diff
@@ -73,6 +75,18 @@ type ErrorResponse struct {
 	IsAPIError bool
 }
 
+func (er ErrorResponse) Error() string {
```

Comment:
I think this should be a pointer type.

I'm not sure what to expect from this function in practice, especially with different combinations of status codes, IsAPIError flags, and the original errors.

Reply:
A table test of this type that demonstrates some of the expected and possible values would help.

Reply:
@displague The panic that this is fixing happens somewhere deep in the TF SDK code. Somehow we pass the ErrorResponse struct to the SDK as an error interface, and when it's about be cast, the code panics because the ErrorResponse doesn't have the Erorr() method. If I recall correctly.

To invoke this error, I had to work with a Metal support eng to set up limit for one of my projects, so that I can hit the situation from #195. When I managed to invoke it and debug the error, it took about a week to remove the project limit (or some other restriction). Right now I can't test it, and I don't know if this will work if we do `func (er *ErrorResponse) Error() string {`. 

As the panic was popping up in TF SDK code, to properly test this we'd need to build the erring response and pass it to initialized provider. That would mean to create a mock API endpoint, and (since the error that this is fixing happens on Delete) mock responses for device creation POST, device resource GET, and finally the DELETE. I wouldn't know where to start with this.

Another way to mitigate #195 without merging this is to fix the API behavior, so that it never sends `errors: []` in a response.

TLDR, I can't verify it this will work if I use pointer type, and I can't imagine a doable way to test this.

Reply:
(as we discussed in a DM) 40bf28c7be0db195afb56b849d09f4f5263f6003 adds mocks to simplify testing of the error conditions.